src/clarity-angular/popover/providers/dropdown.service.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {Injectable} from "@angular/core";
+import {Observable} from "rxjs/Observable";
+import {Subject} from "rxjs/Subject";
+
+@Injectable()
+export class DropdownService {
```

Comment:
Please rename this `RootDropdownService` or something similar. We want to make it clear that it's a singleton per dropdown + nested dropdowns, declared at the root, so that we can move some of the "if root" logic in there.